File: WebApp/wwwroot/scripts/domain/metricFilters/metricFilter.ts
module App.Models {
    "use strict";

    export interface IMetricFilter {
        metricId: string;
        shortTitle: string;
        type: string;
    }

    export interface IMetricFilterOption {
        text: string;
        value: number;
        selected: boolean;
    }

    export const MetricFilterTypes = {
        Text: 'Text',
        Checkbox: 'Checkbox',
        DateRange: 'DateRange',
        NumericRange: 'NumericRange',
        TimeRange: 'TimeRange'
    }

}

File: Models/MetricFilters/MetricFilter.cs
namespace LightMethods.Survey.Models.MetricFilters
{
    public enum MetricFilterTypes
    {
        Text,
        Checkbox,
        DateRange,
        NumericRange,
        TimeRange
    }

    public class MetricFilter
    {
        public string ShortTitle { get; set; }

        public string Type { get; set; }
    }

    public class MetricFilterOption
    {
        public string Text { get; set; }

        public int Value { get; set; }

    }
}
